Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,9 +138,171 @@ public class PushDeploymentController : ApiController
+                            deploymentInfo.TargetFileName = "startup.bat";
+                        }
+                        else
+                        {
+                            deploymentInfo.TargetFileName = "startup.sh";
+                        }
+
+                        break;
+
+                    case Constants.Ear:
+
+                        //currently not supported on Windows but here for future use
+                        if (websiteStack != Constants.JavaEE)
+                        {
+                            return Request.CreateResponse(HttpStatusCode.BadRequest, $"JAR files cannot be deployed to {websiteStack}");
+                        }
+
+                        deploymentInfo.TargetFileName = "app.ear";
+
+                        break;
+
+                    case Constants.Static:
+                        if (string.IsNullOrWhiteSpace(path))
+                        {
+                            return Request.CreateResponse(HttpStatusCode.BadRequest, $"Path must be defined for static file deployments");
+                        }
+                        SetTargetFromPath(deploymentInfo, path);
+
+                        break;
+
+                    case Constants.Zip:
+                        deploymentInfo.Fetch = LocalZipHandler;
+                        if (_settings.RunFromLocalZip())
+                        {
+
+                            // This is used if the deployment is Run-From-Zip
+                            // the name of the deployed file in D:\home\data\SitePackages\{name}.zip is the 
+                            // timestamp in the format yyyMMddHHmmss. 
+                            deploymentInfo.FileName = $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}.zip";
+
+                            // This is also for Run-From-Zip where we need to extract the triggers
+                            // for post deployment sync triggers.
+                            deploymentInfo.SyncFunctionsTriggersPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                        }
+
+                        break;
+
+                    default:
+                        if (string.IsNullOrWhiteSpace(type))
+                        {
+                            //return error: type not provided
+                            return Request.CreateResponse(HttpStatusCode.BadRequest, $"Deployment type not provided");
+                        }
+
+                        //return error: type not valid
+                        return Request.CreateResponse(HttpStatusCode.BadRequest, $"Deployment type {type} not valid");
+                }
+
+                return await PushDeployAsync(deploymentInfo, async, requestObject, type);
+            }
+        }
+
+        private static void SetTargetFromPath(DeploymentInfoBase deploymentInfo, string path)
+        {
+            //When the file name is unknown, then deploymentInfo.FileName must be derived from the path query string
```

Comment:
// Extract directory path and file name from 'path'
// Example: path=a/b/c.jar => TargetDirectoryName=a/b and TargetFileName=c.jar